packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -47,8 +47,218 @@ def _create_file(
+                print(f"{index}. {service_class}")
+
+            selection = click.prompt(
+                "\nSelect a measurement to generate a client",
+                type=int,
+            )
+            measurement_service_class = _get_selected_measurement_service_class(
+                selection, measurement_service_class
+            )
+
+    else:
+        if not measurement_service_class:
+            raise click.ClickException(
+                "The measurement service class cannot be empty. Either provide a measurement service class or use the 'all' flag to generate clients for all registered measurements."
+            )
+    return measurement_service_class
+
+
+def _generate_measurement_client(
+    discovery_client: DiscoveryClient,
+    channel_pool: GrpcChannelPool,
+    service_class: str,
+    built_in_import_modules: List[str],
+    custom_import_modules: List[str],
+    module_name: str,
+    class_name: str,
+    directory_out_path: pathlib.Path,
+) -> None:
+    measurement_service_stub = get_measurement_service_stub(
+        discovery_client, channel_pool, service_class
+    )
+    metadata = measurement_service_stub.GetMetadata(v2_measurement_service_pb2.GetMetadataRequest())
+    configuration_metadata = get_configuration_metadata_by_index(metadata, service_class)
+    output_metadata = get_output_metadata_by_index(metadata)
+
+    configuration_parameters_with_type_and_default_values, measure_api_parameters = (
+        get_configuration_parameters_with_type_and_default_values(
+            configuration_metadata, built_in_import_modules
+        )
+    )
+    output_parameters_with_type = get_output_parameters_with_type(
+        output_metadata, built_in_import_modules, custom_import_modules
+    )
+
+    _create_file(
+        template_name="measurement_plugin_client.py.mako",
+        file_name=f"{module_name}.py",
+        directory_out=directory_out_path,
+        class_name=class_name,
+        display_name=metadata.measurement_details.display_name,
+        configuration_metadata=configuration_metadata,
+        output_metadata=output_metadata,
+        service_class=service_class,
+        configuration_parameters_with_type_and_default_values=configuration_parameters_with_type_and_default_values,
+        measure_api_parameters=measure_api_parameters,
+        output_parameters_with_type=output_parameters_with_type,
+        built_in_import_modules=to_ordered_set(built_in_import_modules),
+        custom_import_modules=to_ordered_set(custom_import_modules),
+    )
+
+    print(
+        f"The measurement plug-in client for the service class '{service_class}' has been created successfully."
+    )
+
+
+def _create_client(
+    channel_pool: GrpcChannelPool,
+    discovery_client: DiscoveryClient,
+    built_in_import_modules: List[str],
+    custom_import_modules: List[str],
+    measurement_service_class: List[str] = [],
+    all: bool = False,
+    module_name: Optional[str] = "",
```

Comment:
Can we change the default values to the optional parameters as None? As the None checks will fail because of this.
```suggestion
    module_name: Optional[str] = None,
    class_name: Optional[str] = None,
    directory_out: Optional[str] = None,
```